vcxagency-node/src/configuration/app-config.js

```diff
@@ -42,6 +42,10 @@ const configValidation = Joi.object().keys({
   LOG_JSON_TO_CONSOLE: Joi.string().valid(['true', 'false']),
   SERVER_PORT: Joi.number().integer().min(1025).max(65535).required(),
   SERVER_MAX_REQUEST_SIZE_KB: Joi.number().integer().min(1).max(MB_AS_KB * 10).required(),
+  SERVER_ENABLE_TLS: Joi.string().valid(['true', 'false']),
+  CERTIFICATE_PATH: Joi.string(),
+  CERTIFICATE_KEY_PATH: Joi.string(),
+  CERTIFICATE_AUTHORITY_PATH: Joi.string(),
```

Comment:
Let's also put additional validation
```
function validateAppConfig (appConfig, callback) {
  Joi.validate(appConfig, configValidation, callback)
  >>> here
}
```
And check that if `SERVER_ENABLE_TLS === 'true'`, `CERTIFICATE_PATH ` and `CERTIFICATE_KEY_PATH ` are specified and files exist.
And if `CERTIFICATE_AUTHORITY_PATH ` is defined, it should also exist as path. 

Unit tests or two for this would be also nice (we have a few for config validation)

Reply:
I thought of that, wasn't sure if it would be worth it, since if `SERVER_ENABLE_TLS === 'true'` and the paths don't exist, the initialization fails with pretty expressive error anyway.